openstack/image/api.go

```diff
@@ -253,6 +267,8 @@ func (h APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 // on return values all the time.
 func errorResponse(err error) APIResponse {
 	switch err {
+	case ErrNoImage:
+		return APIResponse{http.StatusNotFound, nil}
 	default:
```

Comment:
Shouldn't there be a response for ErrImageSaving? I would think this would be a 4xx response and not a 5xx.